routing/routing.go

```diff
@@ -160,11 +166,44 @@ func (m *LBMetrics) DecInflightRequest() {
 	atomic.AddInt64(&m.inflightRequests, -1)
 }
 
-// GetInflightRequests decrements the number of outstanding requests from the proxy to a given backend.
+// GetInflightRequests loads the number of outstanding requests from the proxy to a given backend.
 func (m *LBMetrics) GetInflightRequests() int {
 	return int(atomic.LoadInt64(&m.inflightRequests))
 }
 
+// AddFailedRequest increments the counters related to failed requests and total requests from the proxy to a given backend.
+func (m *LBMetrics) AddFailedRequest() {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	m.failedRequests++
+	m.totalRequests++
+}
+
+func (m *LBMetrics) AddSucceededRequest() {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	m.totalRequests++
```

Comment:
why not 
```
atomic.AddInt64(&m.totalRequests, 1)
```

Reply:
Because in this section of code `m.m` is locked, so we can rely on no other goroutine using those fields.
So both usual and atomic instructions are correct.
However, atomic ones are slower, because they add `mfence` instructions, like here https://stackoverflow.com/questions/51846894/what-is-the-performance-of-stdatomic-vs-non-atomic-variables

Reply:
I guess we discuss it in the comment above https://github.com/zalando/skipper/pull/2363#discussion_r1213458051